flink-core/src/main/java/org/apache/flink/core/fs/FileSystem.java

```diff
@@ -159,56 +164,91 @@ public int hashCode() {
 	/**
 	 * Returns a reference to the {@link FileSystem} instance for accessing the
 	 * local file system.
-	 * 
+	 *
 	 * @return a reference to the {@link FileSystem} instance for accessing the
-	 *         local file system.
+	 * local file system.
 	 */
 	public static FileSystem getLocalFileSystem() {
 		// this should really never fail.
 		try {
 			URI localUri = OperatingSystem.isWindows() ? new URI("file:/") : new URI("file:///");
 			return get(localUri);
-		}
-		catch (Exception e) {
+		} catch (Exception e) {
 			throw new RuntimeException("Cannot create URI for local file system");
 		}
 	}
 
 	/**
+	 * The default filesystem scheme to be used. This can be specified by the parameter
+	 * <code>fs.default-scheme</code> in <code>flink-conf.yaml</code>. By default this is
+	 * set to <code>file:///</code> and uses the local filesystem.
+	 * */
+	private static URI defaultScheme;
+
+	/**
+	 * Sets the default filesystem scheme based on the user-specified configuration parameter
+	 * <code>fs.default-scheme</code>.
+	 * <li>
+	 * As an example, if set to <code>hdfs://localhost:9000/</code>, then an HDFS deployment
+	 * with the namenode being on the local node and listening to port 9000 is going to be used.
+	 * In this case, a file path specified as <code>/user/USERNAME/in.txt</code>
+	 * is going to be transformed into <code>hdfs://localhost:9000/user/USERNAME/in.txt</code>. By
```

Comment:
Depending on what you think about my comments about the doc text, you could also apply it here.